feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/ConversationViewModel.kt

```diff
@@ -715,4 +727,87 @@ internal open class ConversationViewModel(
             imageProvider.createImageLoader(context, authorizationToken)
         }
     }
+
+    fun onFileSelected(uri: Uri?, context: Context) {
+        viewModelScope.launch {
+            _selectedFileUri.emit(uri)
+            val fileName = resolveFileName(context, uri)
+            _selectedFileName.emit(fileName)
+            uploadFileOrImage(context)
+        }
+    }
+
+    private fun uploadFileOrImage(context: Context) {
+        viewModelScope.launch(coroutineContext) {
+            try {
+                val fileBytes = withContext(Dispatchers.IO) {
+                    _selectedFileUri.value?.let { uri ->
+                        context.contentResolver.openInputStream(uri)?.use { inputStream ->
+                            val fileSize = inputStream.available()
+
+                            val maxFileSize = 5 * 1024 * 1024
+                            if (fileSize > maxFileSize) {
+                                throw IllegalArgumentException(getString(context, R.string.conversation_vm_file_size_exceed))
+                            }
+                            inputStream.readBytes()
+                        }
+                    } ?: throw IllegalArgumentException(getString(context, R.string.conversation_vm_file_upload_failed))
+                }
+
+                val response = metisModificationService.uploadFileOrImage(
+                    context = metisContext,
+                    courseId = courseId,
+                    conversationId = conversationId,
+                    fileBytes = fileBytes,
+                    fileName = _selectedFileName.value,
+                    serverUrl = serverConfigurationService.serverUrl.first(),
+                    authToken = accountService.authToken.first()
+                )
+
+                val fileUploadResponse = (response as NetworkResponse.Response).data
+                val filePath = fileUploadResponse.path
+                    ?: throw IllegalArgumentException(getString(context, R.string.conversation_vm_file_upload_failed))
+
+                val currentText = newMessageText.value.text
+                val updatedText = if (isImage(_selectedFileName.value)) {
+                    "$currentText\n![${_selectedFileName.value}]($filePath)\n" // Image markdown
+                } else {
+                    "$currentText\n[${_selectedFileName.value}]($filePath)\n"  // File markdown
```

Comment:
Please use `transformFileUploadMessageMarkdown` from `PostArtemisMarkdownTransformer` for this to not duplicate logic

Reply:
I have update my PR I am not sure for this part I used PostArtemisMarkdownTransformer as you suggested but I am not sure if it is the correct way

Reply:
lgtm 👍